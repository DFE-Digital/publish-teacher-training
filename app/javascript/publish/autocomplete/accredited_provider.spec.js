/**
 * @jest-environment jsdom
 */

import init from './accredited_provider'

describe('Accredited provider', () => {
  describe('init', () => {
    beforeEach(() => {
      document.body.innerHTML = `
         <div id="outer-container">
           <input type="text" id="accredited-provider-search-form-query-field">
           <div id="accredited-provider-autocomplete"></div>
         </div>
       `

      init()
    })

    it('should instantiate an autocomplete', () => {
      expect(document.querySelector('#outer-container')).toMatchSnapshot()
    })
  })
})
