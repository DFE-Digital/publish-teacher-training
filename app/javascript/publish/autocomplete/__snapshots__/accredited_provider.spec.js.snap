// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accredited provider init should instantiate an autocomplete 1`] = `
<div
  id="outer-container"
>
  
           
  <input
    id="old-accredited-provider-search-form-query-field"
    name="provider_id"
    type="hidden"
  />
  
           
  <div
    id="accredited-provider-autocomplete"
  >
    <div
      class="autocomplete__wrapper"
    >
      <div
        style="border: 0px; height: 1px; margin-bottom: -1px; margin-right: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          id="accredited-provider-search-form-query-field__status--A"
          role="status"
        >
          
        </div>
        <div
          aria-atomic="true"
          aria-live="polite"
          id="accredited-provider-search-form-query-field__status--B"
          role="status"
        >
          
        </div>
      </div>
      
      <input
        aria-autocomplete="both"
        aria-describedby="accredited-provider-search-form-query-field__assistiveHint"
        aria-expanded="false"
        aria-owns="accredited-provider-search-form-query-field__listbox"
        autocomplete="off"
        class="autocomplete__input autocomplete__input--default"
        id="accredited-provider-search-form-query-field"
        name=""
        placeholder=""
        role="combobox"
        type="text"
      />
      
      <ul
        class="autocomplete__menu autocomplete__menu--inline autocomplete__menu--hidden"
        id="accredited-provider-search-form-query-field__listbox"
        role="listbox"
      >
        
      </ul>
      <span
        id="accredited-provider-search-form-query-field__assistiveHint"
        style="display: none;"
      >
        When autocomplete results are available use up and down arrows to review and enter to select.  Touch device users, explore by touch or with swipe gestures.
      </span>
    </div>
  </div>
  
         
</div>
`;
